wandb/sdk/artifacts/storage_handlers/s3_handler.py

```diff
@@ -101,7 +101,7 @@ def load_path(
 
         try:
             etag = (
-                self._etag_from_obj(obj_version)
+                obj_version.head()["ETag"][1:-1]  # escape leading and trailing
```

Comment:
does head() return 404 is the object is not found?

Reply:
as per the documentation it should, but im testing that now

Reply:
ok, tested. the allow-missing-refs path works for versioned objects too, confirming that head() returns 404 when the object doesn't exist.